java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSessionFactoryConfig.java

```diff
@@ -4,34 +4,65 @@
 package io.deephaven.client.impl;
 
 import io.deephaven.annotations.BuildableStyle;
+import io.deephaven.barrage.flatbuf.BarrageMessageWrapper;
+import io.deephaven.client.grpc.UserAgentUtility;
 import io.grpc.ManagedChannel;
 import org.apache.arrow.memory.BufferAllocator;
 import org.immutables.value.Value.Default;
 import org.immutables.value.Value.Immutable;
 
+import java.util.Collections;
+import java.util.List;
 import java.util.Objects;
 import java.util.concurrent.ScheduledExecutorService;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 @Immutable
 @BuildableStyle
 public abstract class BarrageSessionFactoryConfig {
     private static final SessionConfig SESSION_CONFIG_EMPTY = SessionConfig.builder().build();
 
+    static final List<String> VERSION_PROPERTIES = Collections.unmodifiableList(Stream.concat(
+            FlightSessionFactoryConfig.VERSION_PROPERTIES.stream(),
+            Stream.of(UserAgentUtility.versionProperty("barrage", BarrageMessageWrapper.class)))
+            .collect(Collectors.toList()));
+
+    private static final ClientChannelFactory CLIENT_CHANNEL_FACTORY = ClientChannelFactoryDefaulter.builder()
+            .userAgent(userAgent(Collections.singletonList("deephaven-java-client-barrage")))
```

Comment:
super nitpicky (so feel free to skip), but this could also just be
```suggestion
            .userAgent(userAgent(List.of("deephaven-java-client-barrage")))
```
and save a handful of chars, another import